lib/active_merchant/billing/gateways/cyber_source_rest.rb

```diff
@@ -245,68 +320,103 @@ def add_merchant_description(post, options)
       end
 
       def url(action)
-        "#{(test? ? test_url : live_url)}/pts/v2/#{action}"
+        case action
+        when /customers/
+          "#{(test? ? test_url : live_url)}/tms/v2/#{action}"
+        when 'instrumentidentifiers', 'paymentinstruments'
+          "#{(test? ? test_url : live_url)}/tms/v1/#{action}"
+        else
+          "#{(test? ? test_url : live_url)}/pts/v2/#{action}"
+        end
       end
 
       def host
         URI.parse(url('')).host
       end
 
       def parse(body)
+        return {} if body.blank?
+
         JSON.parse(body)
       end
 
-      def commit(action, post)
-        response = parse(ssl_post(url(action), post.to_json, auth_headers(action, post)))
+      def commit(action, post, http_method = :post)
+        # headers = http_method == :delete ? auth_headers_delete(action, post) : auth_headers(action, post)
+        headers = auth_headers(action, post, http_method)
+        response = parse(ssl_request(http_method, url(action), post.nil? || post.empty? ? nil : post.to_json, headers))
+        succeeded = success_from(action, response, http_method)
+        body = action == :delete ? { response_code: response.to_s } : response
         Response.new(
-          success_from(response),
-          message_from(response),
+          succeeded,
+          message_from(body, succeeded, http_method),
           response,
-          authorization: authorization_from(response),
+          authorization: authorization_from(response, action),
           avs_result: AVSResult.new(code: response.dig('processorInformation', 'avs', 'code')),
           # cvv_result: CVVResult.new(response['some_cvv_response_key']),
           test: test?,
-          error_code: error_code_from(response)
+          error_code: error_code_from(response, succeeded)
         )
       rescue ActiveMerchant::ResponseError => e
         response = e.response.body.present? ? parse(e.response.body) : { 'response' => { 'rmsg' => e.response.msg } }
         Response.new(false, response.dig('response', 'rmsg'), response, test: test?)
       end
 
-      def success_from(response)
-        %w(AUTHORIZED PENDING REVERSED).include?(response['status'])
+      def success_from(action, response, http_method)
+        return response.empty? if http_method == :delete
```

Comment:
❓ Question:

Are you getting just 200 response without body here ?

Reply:
Yes, response without a body.